hist/hist/inc/TH1.h

```diff
@@ -630,7 +630,11 @@ class TH1F : public TH1, public TArrayF {
    TH1F& operator=(const TH1F &h1);
    ~TH1F() override;
 
-   void     AddBinContent(Int_t bin) override {++fArray[bin];}
+   /// Increment bin content by 1.
+   /// Passing an out-of-range bin leads to undefined behavior
+   void     AddBinContent(Int_t bin) override {if (fArray[bin] < 16777216.f) ++fArray[bin];}
```

Comment:
We could use here a const expr value for the maximum as 
`const expr float bin_content_max = 1./(0.5 * std::numeric_limits<float>::epsilon();`